src/sentry/snuba/discover.py

```diff
@@ -848,16 +858,32 @@ def histogram_query(
         conditions.append([histogram_alias, "<=", max_bin])
 
     columns = [] if key_column is None else [key_column]
-    results = query(
-        selected_columns=columns + [histogram_column, "count()"],
-        conditions=conditions,
-        query=user_query,
-        params=params,
-        orderby=[histogram_alias],
-        limit=len(fields) * num_buckets,
-        referrer=referrer,
-        functions_acl=["array_join", "histogram"],
-    )
+
+    if group_by:
+        results = query(
+            selected_columns=columns + [histogram_column, "count()"],
+            conditions=conditions,
+            query=user_query,
+            params=params,
+            orderby=[histogram_alias],
+            limit=len(fields) * num_buckets,
+            group_by=group_by,
+            referrer=referrer,
+            functions_acl=["array_join", "histogram"],
+        )
+        return results
+
+    else:
+        results = query(
+            selected_columns=columns + [histogram_column, "count()"],
+            conditions=conditions,
+            query=user_query,
+            params=params,
+            orderby=[histogram_alias],
+            limit=len(fields) * num_buckets,
+            referrer=referrer,
+            functions_acl=["array_join", "histogram"],
+        )
```

Comment:
Instead could we refactor this to use `prepare_discover_query` instead? something like
```python
histogram_query = prepare_discover_query(...)
if group_by:
    histogram_query.groupby += group_by
 ```